tests/unit/test_settings_logs.py

```
@@ -41,3 +41,85 @@ def test_api_logs_rate_limit(client, monkeypatch):
+    resp = client.get("/download-logs")
+    assert resp.status_code == 200
+    disposition = resp.headers.get("Content-Disposition", "")
+    match = re.search(r"inkypi_\d{8}-\d{6}\.log", disposition)
+    assert match is not None
+
+
+def test_download_logs_exception_500(client, monkeypatch):
+    """_read_log_lines raising returns 500 text response."""
+    import blueprints.settings as mod
+
+    monkeypatch.setattr(mod, "_read_log_lines", lambda h: (_ for _ in ()).throw(RuntimeError("fail")))
```

Ruff (0.15.7)

[warning] Unused lambda argument: `h`

(ARG005)